components/console_cmd_ping/include/console_ping.h

```diff
@@ -0,0 +1,10 @@
+/*
+ * SPDX-FileCopyrightText: 2023 Espressif Systems (Shanghai) CO LTD
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+__attribute__((weak)) void console_cmd_init(void);
```

Comment:
Have you checked https://github.com/espressif/esp-idf/tree/master/examples/build_system/cmake/plugins#overview-of-link-time-registration ?

Maybe some ideas of link-time registration could be reused here?